src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -255,22 +245,30 @@ class DirectIPaymentPolicy
     // return: first element is max amount that can flow,
     //         second is true if dst holds an iou from src.
     std::pair<IOUAmount, bool>
-    maxFlow (DirectStepI<DirectIPaymentPolicy> const& step,
-        ReadView const& sb, IOUAmount const& desired) const;
+    maxFlow (ReadView const& sb, IOUAmount const& desired) const;
 
-    // Verify the consistency of the step.  These checks are specific to the
-    // policy and assume that non-policy related checks were already performed.
+    // Verify the consistency of the step.  These checks are specific to
+    // payments and assume that general checks were already performed.
     TER
-    check (DirectStepI<DirectIPaymentPolicy> const& step,
-        StrandContext const& ctx,
+    checkSpecifics (StrandContext const& ctx,
```

Comment:
In the name `checkSpecifics`, `Specifics` doesn't add information. I'd just call this `check` (and bring in the base class's `check` through a using so it isn't hidden), or call it `checkHelper` if you don't like the overload.